internal/transactionsubmission/engine/signature_service.go

```diff
@@ -232,38 +233,54 @@ func (ds *DefaultSignatureService) SignFeeBumpStellarTransaction(ctx context.Con
 	return signedFeeBumpStellarTx, nil
 }
 
-func (ds *DefaultSignatureService) BatchInsert(ctx context.Context, kps []*keypair.Full, shouldEncryptSeed bool, currLedgerNumber int) (err error) {
-	if len(kps) == 0 {
-		return fmt.Errorf("no keypairs provided")
+func (ds *DefaultSignatureService) BatchInsert(ctx context.Context, amount int) (publicKeys []string, err error) {
+	if amount < 1 {
+		return nil, fmt.Errorf("the amnount of accounts to insert need to be greater than zero")
+	}
+
+	currentLedgerNumber, err := ds.ledgerNumberTracker.GetLedgerNumber()
+	if err != nil {
+		return nil, fmt.Errorf("getting current ledger number: %w", err)
 	}
+	lockedToLedgerNumber := currentLedgerNumber + IncrementForMaxLedgerBounds
 
 	batchInsertPayload := []*store.ChannelAccount{}
-	for _, kp := range kps {
+	for range make([]interface{}, amount) {
```

Comment:
This syntax is not clear and may lead to misunderstanding.

```suggestion
	for i := 0; i < amount; i++ {
```

Reply:
Done in https://github.com/stellar/stellar-disbursement-platform-backend/pull/165/commits/d1289c36522c84aea1d84f898cc9b679bf5b2c47.